### ISSUE TITLE
New rule: reverse_if_to_reduce_nesting

### ISSUE THREAD

**Issue by danylo-safonov-solid** [solid-software/solid_lints#91]
BAD:
```dart
void x() {
  if (a) {
    if (b) {
      if (c) {
        d;
      }
    }
  }
}
```

GOOD:
```dart
void x() {
  if (!a) return;

  if (!b) return;

  if (!c) return;

  d;
}
```

**Comment by yurii-prykhodko-solid** [comment]
Started on in #154

**Comment by illia-romanenko** [comment]
Fixed in #154
